ci-automation/vendor-testing/azure.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+set -x
+
+# Test execution script for the qemu vendor image.
+# This script is supposed to run in the SDK container.
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/vendor_test.sh
+
+board="${CIA_ARCH}-usr"
+basename="ci-${CIA_VERNUM//+/-}-${CIA_ARCH}"
+azure_instance_type="${AZURE_MACHINE_SIZE}"
+azure_vnet_subnet_name="jenkins-vnet-${AZURE_LOCATION}"
+IS_AMD64=${CIA_ARCH//arm64/}
+
+azure_profile_config_file=''
+secret_to_file azure_profile_config_file "${AZURE_PROFILE}"
+azure_auth_config_file=''
+secret_to_file azure_auth_config_file "${AZURE_AUTH_CREDENTIALS}"
+
+testscript="$(basename "$0")"
+
+# Fetch the Azure image if not present
+if [ -f "${AZURE_IMAGE_NAME}" ] ; then
+    echo "++++ ${testscript}: Using existing ${work_dir}/${AZURE_IMAGE_NAME} for testing ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+else
+    echo "++++ ${testscript}: downloading ${AZURE_IMAGE_NAME} for ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+    copy_from_buildcache "images/${CIA_ARCH}/${CIA_VERNUM}/${AZURE_IMAGE_NAME}.bz2" .
+    cp --sparse=always <(lbzcat "${AZURE_IMAGE_NAME}.bz2") "${AZURE_IMAGE_NAME}"
+    rm "${AZURE_IMAGE_NAME}.bz2"
+fi
+
+if [[ "${CIA_ARCH}" == "arm64" ]]; then
+  AZURE_USE_GALLERY="--azure-use-gallery"
+fi
+
+set -o noglob
```

Comment:
```suggestion
```
Not sure why this is here, maybe it made later glob usage like `merge_tap_files "${main_tapfile}" 'instance_'*'_validate.tap'` fail